<launch>
  <!-- Use simulated time -->
  <param name="use_sim_time" value="true"/>

  <!-- Load the map -->
  <include file="$(find robot_path_planning)/launch/load_map.launch"/>

  <!-- Start Gazebo -->
  <include file="$(find robot_path_planning)/launch/lab.launch" /> 
  
  <!-- Load the robot description -->
  <param name="robot_description" textfile="$(find ugv)/urdf/ugv.urdf" />

  <!-- Spawn the robot in Gazebo -->
  <node name="spawn_ugv" pkg="gazebo_ros" type="spawn_model" args="-urdf -param robot_description -model ugv" />

  <!-- Start joint_state_publisher -->
  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />

  <!-- Start robot_state_publisher -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen" >
    <param name="robot_description" textfile="$(find ugv)/urdf/ugv.urdf" />
  </node>

  <!-- AMCL for Localization -->
  <node name="amcl" pkg="amcl" type="amcl" output="screen">
    <param name="use_map_topic" value="true"/>
    <rosparam file="$(find robot_path_planning)/config/amcl_params.yaml" command="load"/>
  </node> 

  <node pkg="robot_path_planning" type="costmap.py" name="global_costmap" output="screen"/> 

  <node pkg="robot_path_planning" type="global_planner.py" name="global_planner" output="screen"/>

  <node pkg="robot_path_planning" type="local_static.py" name="local_planner" output="screen" />

  <node pkg="robot_path_planning" type="fake_odom.py" name="fake_odom" output="screen"/> 


 <!-- Spawn the diff_drive_controller-->
  <node name="diff_drive_controller" pkg="controller_manager" type="spawner" args="diff_drive_controller" output="screen" >
    <rosparam file="$(find robot_path_planning)/config/controllers.yaml" command="load" />
  </node>

  <!-- RViz for Path Planning -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find robot_path_planning)/rviz/static_path_simulation.rviz" output="screen"/>


</launch>


